test/proposal_test.rb

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require_relative "test_refinements"
+require "storage"
+require "storage/proposal"
+require "storage/fake_probing"
+require "storage/fake_device_factory"
+
+describe Yast::Storage::Proposal do
+  describe "#propose" do
+    using Yast::Storage::TestRefinements
+
+    def input_file_for(name)
+      File.join(DATA_PATH, "input", "#{name}.yml")
+    end
+
+    def output_file_for(name)
+      File.join(DATA_PATH, "output", "#{name}.yml")
+    end
+
+    def fake_scenario(scenario)
+      fake_probing = Yast::Storage::FakeProbing.new
```

Comment:
This line, apart from doing the obvious thing, creates the "fake" entry in the catalog. I have already explained how bad I believe is this. Moreover, creating an empty devicegraph doesn't seem to be a responsibility of `FakeProbing`.